### ISSUE TITLE
Is it possible to access the Send Envelopes more easily?

### ISSUE THREAD

**Issue by Lo-lo78** [jcsteh/osara#558]
Hello,
with the object navigator of NVDA it is possible, with a little attention, to arm the Send Envelopes.
The problem is that when you mix this procedure slows down the work.
Right now I do "shift + l", NVDA + 8 numpad, NVDA + 2 numpad, NVDA + 6 numpad and look for the Send Envelopes.
Is it possible to capture these automations and make a dialog box where the Send Envelopes are, where it is possible to arm or disarm them?
We with OSARA can arm the parameters FX, volume, pan and so on ...
But without NVDA the Send Envelopes cannot be armed.
I thought, for example:
track 1 with a send in track 2;
I select track 1 and with a shortcut linked to an action, a dialog box opens with all the Send Envelopes of the selected track that I can manage more quickly.

1 snare: combo box, "down arrow":
Track 2 "reverb" Send Volume Check box not activated 26 of 42
Track 2 "reverb" Send Pan Check box not activated 29 of 42
Track 2 "reverb" Send Mute Check box not activated 32 of 42

ok button

Obviously if there are two Sends from track 1 to track two it is logical that there will be two combo boxes.
It is also obvious that if there are multiple Sends from track 1 where you go into track 2 and track 3 there will be combo boxes describing the source track and the destination track.
Thanks James!

**Comment by ScottChesworth** [comment]
I'd prefer to see Cockos improve keyboard navigation in Reaper than for OSARA to add another custom dialog. Do you agree, @jcsteh?

**Comment by Lo-lo78** [comment]
Yes, Scott, actually if there were APIs on Reaper for that it would be 
ok. Also the FX Parameter dialog is from OSARA, but as you say Cockos 
better make send envelopes more accessible.

Greetings!

**Comment by jcsteh** [comment]
It would be nice if Cockos improved keyboard navigation in the envelopes dialog, yes. That said, if I were going to do this in OSARA, rather than a new dialog, I'd do it in the Parameters dialog similar to my proposal for envelopes for FX parameters (#533). For example, you'd open the track parameters dialog, move to the send volume parameter and press the Show envelope button.

**Comment by Lo-lo78** [comment]
Hi James, it's a very good way, for example:

2 reverb send volume; Value: progress slider 0.00dB;

Edit Selected 0.00dB;

Arm envelope: Value: progress slider off / on.

! 👍

**Comment by Lo-lo78** [comment]
While you are there you can do this for all parameters in the Track 
Parameters Dialog?

Regards

**Comment by jcsteh** [comment]
Actually, looking at the API, I'm not sure this is even possible. There is a way to retrieve an envelope, but I'm not sure if it creates the envelope if not present.

**Comment by Lo-lo78** [comment]
Yes, I get it, so it's not possible to arm an envelope if it's not 
already activated. Maybe Scott could report it to Justin; it would not 
be bad to open the bees. This is Reaper's strength. Let's see what the 
future will tell us!

**Comment by ElCrackLoko** [comment]
Hello Brother.
Forgive the question, but where do I find these options?

> Track 2 "reverb" Send Volume Check box not activated 26 of 42
> Track 2 "reverb" Send Pan Check box not activated 29 of 42
> Track 2 "reverb" Send Mute Check box not activated 32 of 42
> 
> ok button

Perhaps it is that English is not my native language and I do not finish understanding, but I found this quite interesting. Envelopes with Shift L now that you mention it has given me to investigate (I had seen it before but I don't know why I never moved that lol)
A big greeting and you are well.

**Comment by Lo-lo78** [comment]
Hi, I do this with NVDA's object navigator. Only in this way can I 
activate and arm the envelopes of the sends.

I do it like this:

shift+l, the action is: Track: View envelopes for current/last touched 
track at mouse cursor

Open the envelopes dialog I do, NVDA + 8 numpad, NVDA + 2 numpad and 
then I search with NVDA + 6 numpad, to find: Send Envelopes

If you go ahead with NVDA + 6 numpad, you find:

Track 2 Send Volume  Casella di controllo  non attivato,

Visible  Casella di controllo  non attivato

Arm  Casella di controllo  non attivato

It is necessary to activate only one of these three and arm the envelope.

To arm it you have to bring the mouse to the object navigator with:

NVDA+/ numpad and then left mouse button with / numpad; in this way you 
have activated and armed the envelope.


Sorry when I wrote bees, I meant A P I, that is, Application Programming 
Interface.

Is it possible to notify Cockos if the Application programming 
interfaces for send envelopes can be opened?

Greetings!

**Comment by ElCrackLoko** [comment]
Thank you very much bro. Sure, I was in the right place then, Shift + L. I had been confused because you had talked about shipments and a number of objects that NVDA told you as 32 of 42, 29 of 42, 42 of 42, etc, I thought you meant something else.

**Comment by Lo-lo78** [comment]
It was a copy to clipboard problem, I was not careful!


Il 22/07/2021 20:19, ElCrackLoko ha scritto:

**Comment by jcsteh** [comment]
@ScottChesworth, did you ever raise keyboard navigation in the Envelopes dialog with Cockos, and if so, did you get any traction?

I've figured out how to create envelopes with the API. It's not pretty, but it's doable. Still not going to be trivial to implement, but I understand how it could be done now.

**Comment by ScottChesworth** [comment]
I think I forgot. Sorry chaps. Will get it done today. Still think it's a better approach to try Cockos first.

**Comment by jcsteh** [comment]
> I think I forgot. Sorry chaps. Will get it done today. Still think it's a better approach to try Cockos first.

Theoretically, I agree. That'd mean we'd be using the same interface everyone else does. On the flip side, I worry it might not be as efficient for us unless Cockos implement home/end, page up/page down, first letter navigation, etc. There can be quite a lot of envelopes in that dialog. I guess it depends a) whether they're willing to do it and b) how far they're willing to go.

**Comment by Lo-lo78** [comment]
I close the issue because it has been corrected on Reaper.
